### PR TITLE
VRF-LITE support BGP

### PR DESCRIPTION
**Pull request by Tristan-Labbe**
New pull request with only BGP-table discovery.
For more details, you can see the pull #3255

### CODE REVIEW

**Comment by Tristan-Labbe** [comment]
Hello @laf,

Can you say me if this PR is OK ?

File: includes/discovery/bgp-peers.inc.php [hunk 1]
```diff
@@ -4,231 +4,245 @@
     // Discover BGP peers
     echo 'BGP Sessions : ';
 
-    $bgpLocalAs = trim(snmp_walk($device, '.1.3.6.1.2.1.15.2', '-Oqvn', 'BGP4-MIB', $config['mibdir']));
+    if( key_exists('vrf_lite_cisco', $device) && (count($device['vrf_lite_cisco'])!=0) ){
+        $vrfs_lite_cisco = $device['vrf_lite_cisco'];
+    }
+    else{
+        $vrfs_lite_cisco = array(array('context_name'=>null));
```
Comment (laf) [inline includes/discovery/bgp-peers.inc.php]: I expect this is supposed to cover non vrf_lite setups but It looks like it breaks it. You're creating an empty array and looping though it so no bgp peers will be discovered.

**Comment by Tristan-Labbe** [comment]
Excuse me, I'm new user and I've a question. 
In many script, we can see $device['vrf_lite_cisco'].
But, when I'm looking for the device table on the database, I don't see this column. 
So, can you tell me if it's normal ?

**Comment by Tristan-Labbe** [comment]
This condition  is just to know if the device has vrf. 
If yes, we can know the context name, else we fixed the context name to null.

**Comment by laf** [comment]
I'm not convinced this code does what is intended. You are setting $vrfs_lite_cisco to a string if vrf_lite_cisco is detected and then trying to loop through it. If it's not found vrf_lite_cisco then your setting an array into an array and trying to loop through it.

**Comment by Tristan-Labbe** [comment]
If vrf_lite_cisco is detected, I do that : 
_$vrfs_lite_cisco = $device['vrf_lite_cisco'];_

$device['vrf_lite_cisco'] is not a string, it's an array like that : 

[vrf_lite_cisco] => Array (

    [device_id] => 6
    [hostname] => host
    [sysName] => ...
    [vrf_lite_cisco] => Array
        (

            [0] => Array
                (
                    [vrf_lite_cisco_id] => 87
                    [device_id] => 6
                    [context_name] => Context
                    [intance_name] =>
                    [vrf_name] => context
                )

            [1] => Array
                (
                    [vrf_lite_cisco_id] => 88
                    [device_id] => 6
                    [context_name] =>...
                   )
        )

)



So, if the device has not VRF, we create an array into an array with the context name = null.

**Comment by laf** [comment]
That explains it, thanks.

Just needs a rebase now.

**Comment by Tristan-Labbe** [comment]
Now I think that it's OK.

**Comment by laf** [comment]
Looks good to me.

@librenms/reviewers can we get a second check please?

**Comment by Tristan-Labbe** [comment]
Can you tell me some news about this PR ?

**Comment by Rosiak** [comment]
I'll spin up a test.

**Comment by Tristan-Labbe** [comment]
Thank's

**Pull request merged**